### ISSUE TITLE
Scaffolding other types of files: What filename should I put in?

### ISSUE THREAD

**Issue by doggy8088** [dotnet/EntityFramework.Docs#4456]
In the `Scaffolding other types of files`, you provided an T4 example, but not mentioned about the filename and where should I put it into. I don't know how to use this Mermaid T4 script.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: cae415b7-c0ea-4d96-9d17-7e7d3163c09d
* Version Independent ID: cae415b7-c0ea-4d96-9d17-7e7d3163c09d
* Content: [Custom Reverse Engineering Templates - EF Core](https://learn.microsoft.com/en-us/ef/core/managing-schemas/scaffolding/templates?WT.mc_id=DT-MVP-4015686&tabs=dotnet-core-cli)
* Content Source: [entity-framework/core/managing-schemas/scaffolding/templates.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/managing-schemas/scaffolding/templates.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @bricelam
* Microsoft Alias: **avickers**

**Comment by bricelam** [comment]
The mermaid script is intended to be used as a custom `DbContext.t4`. We should mention this in the article.

**Comment by bricelam** [comment]
You can't, but you can create a separate dummy project for the diagrams. Or just rename the files before generating.

**Comment by ErikEJ** [comment]
You can with EF Core Power Tools

**Comment by doggy8088** [comment]
@ErikEJ Does EFCPT can handle multiple T4 templates other than `DbContext.t4` and `EntityType.t4`?

**Comment by doggy8088** [comment]
The mermaid sample from the doc seems not functional.

![image](https://github.com/dotnet/EntityFramework.Docs/assets/88981/b22fefb7-3df9-47be-86ad-b60481e1792c)

**Comment by ErikEJ** [comment]
@doggy8088 Yes, but the VS and CLI editions support that, [as documented here](https://github.com/ErikEJ/EFCorePowerTools/wiki/Reverse-Engineering#customize-code-using-t4-templates)

**Comment by ErikEJ** [comment]
>> The mermaid sample from the doc seems not functional.

A limitation in the MarkDown viewer?

**Comment by doggy8088** [comment]
> > > The mermaid sample from the doc seems not functional.
> 
> A limitation in the MarkDown viewer?

No, it's a bug. I fixed as below:

````
<#@ output extension=".md" #>
<#@ assembly name="Microsoft.EntityFrameworkCore" #>
<#@ assembly name="Microsoft.EntityFrameworkCore.Relational" #>
<#@ assembly name="Microsoft.EntityFrameworkCore.Design" #>
<#@ parameter name="Model" type="Microsoft.EntityFrameworkCore.Metadata.IModel" #>
<#@ parameter name="Options" type="Microsoft.EntityFrameworkCore.Scaffolding.ModelCodeGenerationOptions" #>
<#@ import namespace="System.Linq" #>
<#@ import namespace="Microsoft.EntityFrameworkCore" #>
# <#= Options.ContextName #>

```mermaid
erDiagram
<#
    foreach (var entityType in Model.GetEntityTypes().Where(e => !e.IsSimpleManyToManyJoinEntityType()))
    {
#>
    <#= entityType.Name #> {
    }
<#
        foreach (var foreignKey in entityType.GetForeignKeys())
        {
#>
    <#= entityType.Name #> <#= foreignKey.IsUnique ? "|" : "}" #>o--<#= foreignKey.IsRequired ? "|" : "o" #>| <#= foreignKey.PrincipalEntityType.Name #> : "<#= foreignKey.GetConstraintName() #>"
<#
        }

        foreach (var skipNavigation in entityType.GetSkipNavigations().Where(n => n.IsLeftNavigation()))
        {
#>
    <#= entityType.Name #> }o--o{ <#= skipNavigation.TargetEntityType.Name #> : <#= skipNavigation.JoinEntityType.Name #>
<#
        }
    }
#>
```
````

**Comment by ErikEJ** [comment]
Maybe a docs PR?

**Comment by doggy8088** [comment]
> @doggy8088 Yes, both the VS and CLI editions support that, [as documented here](https://github.com/ErikEJ/EFCorePowerTools/wiki/Reverse-Engineering#customize-code-using-t4-templates)
@ErikEJ 
I put a `Docs/CodeTemplates/EFCore/DbContext.t4` file. How can I let `ContosoUniversityContext.md` into `Docs/` folder?

![image](https://github.com/dotnet/EntityFramework.Docs/assets/88981/111c8969-bb88-48b3-b4c8-6ee589773478)

**Comment by ErikEJ** [comment]
>>  How can I let ContosoUniversityContext.md into Docs/ folder?

@doggy8088 Currently the files generated by DbContext.t4 are placed in same location as DbContext.cs, but I think it could be changed to the Docs folder. Please create an issue in EF Core Power Tools repo and I will investigate.
